### PR TITLE
Adding required attribute to text and textarea form fields

### PR DESCRIPTION
**Pull request by multiwebinc**
For client-side form field validation. See: https://github.com/octobercms/october/pull/4804

### CODE REVIEW

File: formwidgets/mltextarea/partials/_mltextarea.htm [hunk 1]
```diff
@@ -15,7 +15,8 @@
             placeholder="<?= e(trans($field->placeholder)) ?>"
             class="form-control field-textarea size-<?= $field->size ?>"
             autocomplete="off"
-            <?= $field->getAttributes() ?>><?= e($field->value) ?></textarea>
+            <?= $field->getAttributes() ?>
+            <?= $field->required ? 'required' : '' ?>><?= e($field->value) ?></textarea>
```
Comment (mjauvin) [inline formwidgets/mltextarea/partials/_mltextarea.htm:19]:
```suggestion
            <?= $field->required ? 'required' : '' ?>>
            <?= e($field->value) ?>
        </textarea>
```

**Comment by mjauvin** [comment]
Can you give a bit more details why this change is needed?

File: formwidgets/mltextarea/partials/_mltextarea.htm (hunk 1 above)
Comment (multiwebinc) [reply formwidgets/mltextarea/partials/_mltextarea.htm:19]: Literally anything (including line breaks and white space) between the `<textarea></textarea>` tags will be added to the rendered element. This is not what you want.

**Comment by multiwebinc** [comment]
This allows the "required" attribute of the form element to be validated client-side. See https://developer.mozilla.org/en-US/docs/Learn/Forms/Form_validation.

File: formwidgets/mltextarea/partials/_mltextarea.htm (hunk 1 above)
Comment (mjauvin) [reply formwidgets/mltextarea/partials/_mltextarea.htm:19]: You're right, I had a brain cramp while focusing on coding style... :)

**Comment by multiwebinc** [comment]
Maybe just wait to see if https://github.com/octobercms/october/pull/4804 is merged first. I asked a question there that could make this PR redundant.

**Comment by LukeTowers** [comment]
This PR is redundant, we will have it implemented in the `getAttributes()` method instead.

**Pull request closed without merging**